src/main/java/org/template/Algorithm.java

```diff
@@ -356,23 +356,38 @@ public NullModel calcAll(SparkContext sc, PreparedData preparedData) {
 
     @Override
     public NullModel train(SparkContext sc, PreparedData preparedData) {
-        if (this.recsModel.equals(RecsModel.All) ||
-                this.recsModel.equals(RecsModel.BF)) {
```

Comment:
Replying for the purpose of the thread... There are three cases and three function calls: calcAll with calcPopular being both True and False, and calcPop.